crates/bevy_transform/src/components/children.rs

```diff
@@ -21,20 +25,88 @@ impl MapEntities for Children {
 }
 
 impl Children {
-    pub fn with(entity: &[Entity]) -> Self {
-        Self(SmallVec::from_slice(entity))
+    pub fn with(entities: &[Entity]) -> Self {
+        let mut children = Self::default();
+        for entity in entities {
+            children.push(*entity);
+        }
+        children
     }
 
     /// Swaps the child at `a_index` with the child at `b_index`
     pub fn swap(&mut self, a_index: usize, b_index: usize) {
-        self.0.swap(a_index, b_index);
+        let a_entity = self.order[a_index];
+        let b_entity = self.order[b_index];
+        self.order.swap(a_index, b_index);
+        self.uniqueness.insert(a_entity, b_index);
+        self.uniqueness.insert(b_entity, a_index);
+    }
+
+    pub(crate) fn push(&mut self, entity: Entity) {
+        let order = &mut self.order;
+        let uniqueness = &mut self.uniqueness;
+
+        uniqueness.entry(entity).or_insert_with(|| {
+            order.push(entity);
+            order.len() - 1
+        });
+
+        let desired_index = order.len() - 1;
+        let current_index = uniqueness[&entity];
+        if current_index != desired_index {
+            self.swap(current_index, desired_index);
```

Comment:
Example order is [1,2,3,4] and I push(2). Would the result be [1,4,3,2]?

Reply:
I would need to see some tests for this Children component. It is not entirely trivial anymore.

Reply:
There are a couple of new tests in bevy_transform/src/hierarchy/child_builder.rs, but you are correct they should be closer to the actual code.

Reply:
Also, you are right about the order. My goal is to have the entity be in the same place in the list whether or not it already existed in the set. So if you reinsert an existing entity it will get moved to a new position. I'm not certain that that's the right behavior, I think you can make a strong case for the entity retaining it's old position.

Reply:
I think its good to move the existing entity to the new position, but this should not jump an unrelated entity to another position. In the example I push(2) to the end but 4 jumps from the end to index 1 were the original entity 2 was before to make room for the new entity 2.